### PR TITLE
feat(zk toolbox): External node support

### PR DESCRIPTION
**Pull request by Deniallugo**
## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.

### CODE REVIEW

File: zk_toolbox/crates/zk_supervisor/src/dals.rs [hunk 1]
```diff
@@ -46,7 +46,7 @@ pub fn get_prover_dal(shell: &Shell) -> anyhow::Result<Dal> {
 
     Ok(Dal {
         path: PROVER_DAL_PATH.to_string(),
-        url: secrets.database.prover_url.clone(),
+        url: secrets.database.prover_url.unwrap().clone(),
```
Comment (matias-gonz) [inline zk_toolbox/crates/zk_supervisor/src/dals.rs:49]: Maybe use expect?

File: zk_toolbox/crates/zk_inception/src/commands/external_node/args/run.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+use serde::{Deserialize, Serialize};
+
+use crate::messages::{MSG_SERVER_ADDITIONAL_ARGS_HELP, MSG_SERVER_COMPONENTS_HELP};
+
+#[derive(Debug, Serialize, Deserialize, Parser)]
+pub struct RunExternalNodeArgs {
+    #[clap(long)]
+    pub reinit: bool,
+    #[clap(long, help = MSG_SERVER_COMPONENTS_HELP)]
+    pub components: Option<Vec<String>>,
```
Comment (matias-gonz) [inline zk_toolbox/crates/zk_inception/src/commands/external_node/args/run.rs:11]: Could use `enum` and `clap::ValueEnum`

Comment (Deniallugo) [reply zk_toolbox/crates/zk_inception/src/commands/external_node/args/run.rs:11]: I'm not sure we have to have this dependency. we are adding components back and force constantly

**Review by aon — DISMISSED** [review]

**Review by onyxet — APPROVED** [review]

File: .github/workflows/ci-zk-toolbox-reusable.yml [hunk 3]
```diff
@@ -90,13 +90,31 @@ jobs:
...
 
       - name: Run integration tests
         run: |
           ci_run zk_supervisor integration-tests --ignore-prerequisites --verbose
 
+
+      - name: Run external node server
+        run: |
+          ci_run zk_inception external-node configs --db-url=postgres://postgres:notsecurepassword@postgres:5432 \
+          --db-name=zksync_en_localhost_era  --l1-rpc-url=http://reth:8545
```
Comment (hatemosphere) [inline .github/workflows/ci-zk-toolbox-reusable.yml:104]: redundant space

File: .github/workflows/ci-zk-toolbox-reusable.yml [hunk 4]
```diff
@@ -90,13 +90,31 @@ jobs:
...
       - name: Run integration tests
         run: |
           ci_run zk_supervisor integration-tests --ignore-prerequisites --verbose
 
+
+      - name: Run external node server
+        run: |
+          ci_run zk_inception external-node configs --db-url=postgres://postgres:notsecurepassword@postgres:5432 \
+          --db-name=zksync_en_localhost_era  --l1-rpc-url=http://reth:8545  
+          ci_run zk_inception external-node init  --ignore-prerequisites
```
Comment (hatemosphere) [inline .github/workflows/ci-zk-toolbox-reusable.yml:105]: redundant space

File: .github/workflows/ci-zk-toolbox-reusable.yml [hunk 5]
```diff
@@ -90,13 +90,31 @@ jobs:
...
+          ci_run zk_inception external-node run --ignore-prerequisites  &>external_node.log &
```
Comment (hatemosphere) [inline .github/workflows/ci-zk-toolbox-reusable.yml:106]: redundant space

File: .github/workflows/ci-zk-toolbox-reusable.yml [hunk 6]
```diff
@@ -90,13 +90,31 @@ jobs:
...
+          ci_run sleep 5
+
+      - name: Run integration tests en
+        run: |
+          ci_run zk_supervisor integration-tests --ignore-prerequisites --verbose --external-node
+
+
```
Comment (hatemosphere) [inline .github/workflows/ci-zk-toolbox-reusable.yml:113]: redundant space

**Review by hatemosphere — CHANGES_REQUESTED** [review]

**Review by hatemosphere — APPROVED** [review]

**Pull request merged**